src/structurizr/view/dynamic_view.py

```diff
@@ -0,0 +1,317 @@
+            dynamic_view.add(container2, container1, "Sends response back to")
+        """
+        self.check_element_can_be_added(source)
+        self.check_element_can_be_added(destination)
+        relationship, response = self._find_relationship(
+            source, description, destination, technology
+        )
+        if relationship is None:
+            if technology:
+                raise ValueError(
+                    f"A relationship between {source.name} and "
+                    f"{destination.name} with technology "
+                    f"'{technology}' does not exist in model."
+                )
+            else:
+                raise ValueError(
+                    f"A relationship between {source.name} and "
+                    f"{destination.name} does not exist in "
+                    "model."
+                )
+        self._add_element(source, False)
+        self._add_element(destination, False)
+        return self._add_relationship(
+            relationship,
+            description=description or relationship.description,
+            order=self.sequence_number.get_next(),
+            response=response,
+        )
+
+    @contextmanager
+    def subsequence(self):
+        """
+        Start a subsueqnce through a `with` block.
+
+        Subsequences allow nested interaction sequences, showing "child" calls through
+        numbering 1.1, 1.2 etc.  Subsequences can themselves be nested.
+
+        As an example, assume four Components, A-D.  A makes a service request to B,
+        which in turn calls both C then D to process the request before returning the
+        results back to A.  This can be shown using:
+
+            dynamic_view.add(a, b, "Sends service request to)
+            with dynamic_view.subsequence():
+                dynamic_view.add(b, c, "Makes subcall to")
+                dynamic_view.add(b, d, "Makes second subcall to")
+            dynamic_view.add(b, a, "Sends results back to)
+
+        This would result in four interactions shown, with orders "1", "1.1", "1.2"
+        and "2" respectively.
+        """
+        try:
+            self.sequence_number.start_subsequence()
+            yield self
+        finally:
+            self.sequence_number.end_subsequence()
+
+    @contextmanager
+    def parallel_sequence(self, continue_numbering: bool):
+        r"""
+        Start a parallel sequence through a `with` block.
+
+        Args:
+            continue_numbering: if `True` then when the with block completes, the main
+                                sequence number will continue from after the last
+                                number from the parallel sequence.  If `False` then it
+                                will reset back to the start (usually so you can start
+                                a new parallel sequence).
+
+        Parallel sequences allow for multiple parallel flows to share the same
```

Comment:
I'd use the `Examples:` section again. That means indenting everything.

```suggestion
        Examples:
        Parallel sequences allow for multiple parallel flows to share the same
```